amlb/utils/process.py

```diff
@@ -153,9 +172,20 @@ def read_pipe(pipe, timeout):
                 reads[i] = line
         return reads if len(pipes) > 1 else reads[0]
 
-    output, error = zip(*iter(lambda: read_pipe([process.stdout if process.stdout else 1,
-                                                 process.stderr if process.stderr else 2], activity_timeout),
-                              ['', '']))
+    process_output = list(iter(
+        lambda: read_pipe(
+            [process.stdout if process.stdout else 1,
+             process.stderr if process.stderr else 2
+             ], activity_timeout
+    ), ['', '']))
+    if not process_output:
+        log.warning(
```

Comment:
When tracking process output, if no output is provided this can be an issue with logging. However, depending on the configuration and the integration script, it is also possible that no logging output is the expected scenario. Instead of always giving a warning like this patch does, we should try to inspect the logging configuration (and possibly even integration script) to check whether or not a warning should be give, and what the right warning is.